TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODGameshowBuzzerHostPlugin.cs

```diff
@@ -0,0 +1,927 @@
+                        break;
+                    }
+                case VMEODGameshowBuzzerStates.Disabled:
+                    {
+                        _AnsweringPlayerIndex = -1;
+                        BroadcastToPlayers("BuzzerEOD_Master", new byte[] { 0 }, false);
+                        _Tock = 0;
+                        break;
+                    };
+                case VMEODGameshowBuzzerStates.Locked:
+                    {
+                        // other players should react
+                        var players = GetConnectedPlayers(_AnsweringPlayerIndex, true);
+                        foreach (var otherPlayers in players)
+                            otherPlayers.ActivateOtherBuzzer();
+                        break;
+                    }
+            }
+            _BuzzerState = newState;
+        }
+        private void SetUTimer(int newTime)
+        {
+            _UbiqitousTimer = newTime;
+            BroadcastToPlayers("BuzzerEOD_Timer", _UbiqitousTimer + "", false);
+            MyClient.Send("BuzzerEOD_Timer", _UbiqitousTimer + "");
+        }
+        private void BroadcastToPlayers(string evt, string args, bool onlyActiveBuzzers)
+        {
+            var players = GetConnectedPlayers(-1, onlyActiveBuzzers);
+            foreach (var player in players)
+            {
+                player?.Send(evt, args);
+            }
+        }
+        private void BroadcastToPlayers(string evt, byte[] args, bool onlyActiveBuzzers)
+        {
+            var players = GetConnectedPlayers(-1, onlyActiveBuzzers);
+            foreach (var player in players)
+            {
+                player?.Send(evt, args);
+            }
+        }
+        /// <summary>
+        /// Note: This occurs under locks: ConnectedContestants and PlayerConfigLock
+        /// </summary>
+        /// <param name="playerToSkip"></param>
+        /// <returns></returns>
+        private List<VMEODGameshowBuzzerPlayerPlugin> GetConnectedPlayers(int playerIndexToSkip, bool onlyActiveBuzzers)
+        {
+            var players = new List<VMEODGameshowBuzzerPlayerPlugin>();
+            for (int index = 0; index < 4; index++)
+            {
+                if (index == playerIndexToSkip)
+                    continue;
+                var player = ConnectedContestants[index];
+                if (player != null)
+                {
+                    if (player.EODType.Equals(VMEODGameshowBuzzerPluginType.Player))
+                    {
+                        if (!onlyActiveBuzzers || onlyActiveBuzzers && player.MyBuzzerEnabled)
+                            players.Add(player);
+                    }
+                }
+            }
+            return players;
+        }
+        private int GetPlayerIndex(VMEODGameshowBuzzerPlayerPlugin player)
+        {
+            int playerIndex = -1;
+            lock (PlayerConfigLock)
```

Comment:
Nested locks always give me super anxiety. If you do this, make double sure that they are always locked in the same order, otherwise you can encounter a deadlock.